gbn/go.mod

```diff
@@ -0,0 +1,36 @@
+module github.com/lightninglabs/lightning-node-connect/gbn
```

Comment:
I think we might get away without this commit and make our lives a bit easier in the future.
Doing a quick search, it looks like we only use the `litclient` and `perms` packages from the `lightning-terminal` repository. And both of these are now their own modules.
So requiring the whole `lightning-node-connect` package from the WASM client should no longer pull in all of litd.

Reply:
This commit also gets rid of the `github.com/lightninglabs/lightning-node-connect v0.3.3-alpha.0.20250303090341-f823ad04b9de` dep from WASM-client. We could just replace to the relative parent directory for lnc. But it will block us from importing taproot-assets into any part of LNC in the future.